drizzlepac/hlautils/analyze.py

```diff
@@ -0,0 +1,265 @@
+    catalogSources = 0 # Number of astrometric catalog sources determined based upon coordinate overlap with image WCS
+    foundSources = 0   # Number of sources detected in images
+    matchSources = 0   # Number of sources cross matched between astrometric catalog and detected in image
+    rms_x = -1.0
+    rms_y = -1.0
+    rms_ra = -1.0
+    rms_dec = -1.0
+    chisq_x = -1.0
+    chisq_y = -1.0
+    completed = False # If true, there was no exception and the processing completed all logic
+    dateObs = None     # Human readable date
+    mjdutc = -1.0      # MJD UTC start of exposure
+    fgslock = None
+    processMsg = None
+    status = 9999
+    compromised = 0
+    headerletFile = None
+
+    fit_rms = -1.0
+    total_rms = -1.0
+    datasetKey = -1.0
+
+    namesArray = ('imageName', 'instrument', 'detector', 'filter', 'aperture', 'obstype',
+            'subarray', 'dateObs', 'mjdutc', 'doProcess', 'processMsg', 'catalog', 'foundSources',
+            'catalogSources','matchSources', 'rms_x', 'rms_y', 'rms_ra', 'rms_dec', 'completed',
+            'fit_rms', 'total_rms', 'datasetKey', 'status', 'headerletFile')
+    dataType = ('S20', 'S20', 'S20', 'S20', 'S20', 'S20', 'b', 'S20', 'f8', 'b', 'S30',
+            'S20', 'i4', 'i4', 'i4', 'f8', 'f8', 'f8', 'f8', 'b', 'f8', 'f8', 'i8', 'i4', 'S30')
+
+    # Create an astropy table
+    outputTable = Table(names=namesArray,dtype=dataType)
+
+    # Loop over the list of images to determine viability for alignment processing
+    #
+    # Capture the data characteristics before any evaluation so the information is
+    # available for the output table regardless of which keyword is used to 
+    # to determine the data is not viable for alignment.
+
+    for inputFile in inputFileList:
+
+        header_hdu  = 0
+        header_data = getheader(inputFile, header_hdu)
+
+        # Keywords to use potentially for downstream analysis
+        instrume = (header_data['INSTRUME']).upper()
+        detector = (header_data['DETECTOR']).upper()
+        subarray = header_data['SUBARRAY']
+        dateObs  = header_data['DATE-OBS']
+        mjdutc  = header_data['EXPSTART']
+
+        # Obtain keyword values for analysis of viability
+        obstype  = (header_data[OBSKEY]).upper()
+        mtflag   = (header_data[MTKEY]).upper()
+        
+        scan_typ = ''
+        if instrume == 'WFC3':
+            scan_typ = (header_data[SCNKEY]).upper()
+
+        sfilter = ''
+        if instrume == 'WFC3':
+            sfilter  = (header_data[FILKEY]).upper()
+        # Concatenate the two ACS filter names together with an underscore
+        # If the filter name is blank, skip it
+        if instrume == 'ACS':
+            for filtname in acsFiltNameList:
+
+                # The filter keyword value could be zero or more blank spaces 
+                # Strip off any leading or trailing blanks
+                if len(header_data[filtname].upper().strip()) > 0:
```

Comment:
There is no need to convert a string to upper case in order to check whether it is all-white-space. A simple version would be `if header_data[filtname].strip():`